File: golang-backend/internal/handlers/profilehandler/dto.go
package profilehandler

import (
	"github.com/aportela/doneo/internal/handlers/userhandler"
)

type updateRequest struct {
	Name     string  `json:"name"`
	Email    string  `json:"email"`
	Password *string `json:"password,omitempty"`
}

type profileResponse struct {
	userhandler.UserBaseResponse
	Email     string `json:"email"`
	CreatedAt int64  `json:"createdAt"`
	UpdatedAt *int64 `json:"updatedAt"`
}

File: vite-vue-ts-frontend/src/modules/profile/types/dto.ts
export type UpdateRequest = {
  name: string;
  email: string;
  password?: string;
};

export type SaveAvatarRequest = {
  svg: string;
};

export type ProfileResponse = {
  id: string;
  name: string;
  email: string;
  createdAt: number;
  updatedAt: number | null;
};

export type EmptyResponse = {};
